src/rp2_common/pico_cyw43_driver/cyw43_driver.c

```diff
@@ -104,6 +114,73 @@ static void cyw43_sleep_timeout_reached(async_context_t *context, __unused async
 }
 
 bool cyw43_driver_init(async_context_t *context) {
+#if CYW43_USE_PARTITION_FIRMWARE
+    const int buf_words = (16 * 4) + 1;   // maximum of 16 partitions, each with maximum of 4 words returned, plus 1
+    uint32_t* buffer = malloc(buf_words * 4);
```

Comment:
malloc is best avoided for Micropython (we use some for BT but we need to fix that). You could use cyw43_state as a temp buffer it's not needed yet?

Reply:
Should I switch this one to use cyw43_state and leave the other one with malloc, or just leave them both as malloc?

Reply:
Leave it as it is for now. I can't immediately spot the code in Micropython which controls the heap space we have available.

Reply:
i would just put this on the stack, it is only 17 words